test/elm/aggregate/aggregate-test.ts

```diff
@@ -12,17 +12,17 @@ describe('Count', () => {
     setup(this, data);
   });
 
-  it('should be able to count lists without nulls', function () {
-    this.not_null.exec(this.ctx).should.equal(5);
+  it('should be able to count lists without nulls', async function () {
+    (await this.not_null.exec(this.ctx)).should.equal(5);
   });
-  it('should be able to count lists with nulls', function () {
-    this.has_null.exec(this.ctx).should.equal(2);
+  it('should be able to count lists with nulls', async function () {
+    (await this.has_null.exec(this.ctx)).should.equal(2);
   });
-  it('should be able to count empty list', function () {
-    this.empty.exec(this.ctx).should.equal(0);
+  it('should be able to count empty list', async function () {
+    (await this.empty.exec(this.ctx)).should.equal(0);
   });
-  it('should be able to count null list', function () {
-    this.is_null.exec(this.ctx).should.equal(0);
+  it('should be able to count null list', async function () {
+    (await this.is_null.exec(this.ctx)).should.equal(0);
   });
```

Comment:
Yeah I'm not opposed to this! There's also:

``` TypeScript
return this.not_null.exec(this.ctx).should.be.resolvedWith(5);
```

Which compares the value using deep equality, so that fixes the stack trace issue and avoids the need for async await pattern in tests? That would also require changing the assertion everywhere just for the sake of using Promises but idk

Reply:
https://shouldjs.github.io/#assertion-fulfilledwith

Reply:
I totally forgot about that style assertion.  Good call.  It's not as succinct as my handy little functions, but I like that it will trace to the right place and it's probably a more correct way of using the framework.  I don't think we should do anything right now (unless you have some crazy emacs thing that can do it in > 1 hr) -- but let's keep this in mind.

Reply:
I'm not saying I don't have the capability to use a macro that could accomplish this..... but I agree it probably makes sense to hold off and keep in mind for future refactors of the code